Packs/CheckPointSandBlast/Integrations/CheckPointSandBlast/CheckPointSandBlast.py

```diff
@@ -0,0 +1,927 @@
+
+    interval_in_secs = int(args.get('interval_in_seconds', DEFAULT_INTERVAL))
+    timeout_in_seconds = int(args.get('timeout_in_seconds', DEFAULT_TIMEOUT))
+
+    if 'file_hash' not in args:
+        command_results = upload_function(client, args)
+        outputs = command_results.raw_response
+
+        file_name = dict_safe_get(outputs, ['response', 'file_name'])
+        file_hash = dict_safe_get(outputs, ['response', 'md5'])
+        label = dict_safe_get(outputs, ['response', 'status', 'label'])
+
+        if label not in ('FOUND', 'PARTIALLY_FOUND'):
+            scheduled_command = schedule_next_polling(
+                args=args,
+                command=command,
+                interval_in_secs=interval_in_secs,
+                timeout_in_seconds=timeout_in_seconds,
+                file_name=file_name,
+                file_hash=file_hash
+            )
+            command_results.scheduled_command = scheduled_command
+
+            return command_results
+
+        args['file_name'] = file_name
+        args['file_hash'] = file_hash
+
+    command_results = query_function(client, args)
+    outputs = command_results.raw_response
+    label = dict_safe_get(outputs, ['response', 'status', 'label'])
+
+    if label not in ('FOUND', 'PARTIALLY_FOUND'):
+        scheduled_command = schedule_next_polling(
+            args=args,
+            command=command,
+            interval_in_secs=interval_in_secs,
+            timeout_in_seconds=timeout_in_seconds,
+            file_name=file_name,
+            file_hash=file_hash
+        )
+
+        # result with scheduled_command only - no update to the war room
+        command_results = CommandResults(scheduled_command=scheduled_command)
+
+    return command_results
+
+
+def schedule_next_polling(
+    args: Dict[str, Any],
+    command: str,
+    interval_in_secs: int,
+    timeout_in_seconds: int,
+    file_name: str,
+    file_hash: str,
+) -> ScheduledCommand:
+    """
+    Initialize a scheduled command for the next poll.
+
+    Args:
+        args (Dict[str, Any]): Arguments passed by the user and for the next polling command.
+        command (str): Next command to poll.
+        interval_in_secs (int): Time interval between each polling.
+        timeout_in_seconds (int): Timeout for polling to end.
+        file_name (str): Name of the file to poll.
+        file_hash (str): Hash of the file to poll.
+
+    Returns:
+        ScheduledCommand: Next poll.
+    """
+    polling_args = {
+        'file_name': file_name,
+        'file_hash': file_hash,
+        'interval_in_seconds': interval_in_secs,
+        'polling': True,
+        **args
+    }
+
+    return ScheduledCommand(
+        command=command,
+        next_run_in_seconds=interval_in_secs,
+        args=polling_args,
+        timeout_in_seconds=timeout_in_seconds
+    )
+
+
+''' HELPER FUNCTIONS '''
+
+
+def get_analysis_context_output(output: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
So from what I understand the function should return another output for the fields it can fill in "File" and also update the YML context outputs accordingly?